synapseclient/core/cache.py

```diff
@@ -281,22 +280,34 @@ def _cache_dirs(self):
                     if os.path.isdir(path2) and re.match('\\d+', item2):
                         yield path2
 
-    def purge(self, before_date, dry_run=False):
+    def purge(self, before_date=None, after_date=None, dry_run=False):
         """
         Purge the cache. Use with caution. Delete files whose cache maps were last updated prior to the given date.
 
         Deletes .cacheMap files and files stored in the cache.cache_root_dir, but does not delete files stored outside
         the cache.
         """
+        if before_date is None and after_date is None:
+            raise ValueError("Either before date or after date should be provided")
+
         if isinstance(before_date, datetime.datetime):
             before_date = utils.to_unix_epoch_time_secs(before_date)
+        if isinstance(after_date, datetime.datetime):
+            after_date = utils.to_unix_epoch_time_secs(after_date)
+
+        if before_date and after_date and before_date > after_date:
+            raise ValueError("Before date can not be larger than after date")
```

Comment:
Is this backwards? Shouldn't we allow before_date > after_date but not the reverse?

If we set before_date to datetime.datetime(2021, 2, 9) and after_date to datetime.datetime(2021, 2, 7) then I think we'd want to delete everything between 2/7 and 2/9, which seems like we should allow. If we allow after_date > before_date then it will purge everything except what is between those two dates, which seems a bit confusing.